Realm/TestUtils/include/RLMAssertions.h

```diff
@@ -118,6 +118,44 @@ FOUNDATION_EXTERN bool RLMHasCachedRealmForPath(NSString *path);
     XCTAssertEqual([exception.userInfo[NSUnderlyingErrorKey] code], expectedCode, __VA_ARGS__); \
 })
 
+#define RLMValidateError(error, errDomain, errCode, msg) do {                                           \
```

Comment:
What are the advantages of defining this as a macro than as a function? in TestCases for example.

Reply:
These are macros so that they can report the failing line as the line containing the `RLMValidateError()` call rather than the line within the test function which made the failing assertion. We can use regular functions in Swift because the Swift version of `XCTAssert()` is itself just a function which takes a line number as an argument, so we can pass that along explicitly. In obj-c `XCTAssert()` is a macro which uses `__LINE__` internally, so anything which wraps it must also be macros for `__LINE__` and `__FILE__` to have the correct values.

Reply:
Ohh that's good to know, thanks